go/cmd/vtbench/vtbench.go

```diff
@@ -86,89 +83,97 @@ import (
 */
 
 var (
-	// connection flags
-	host       = flag.String("host", "", "vtgate host(s) in the form 'host1,host2,...'")
-	port       = flag.Int("port", 0, "vtgate port")
-	unixSocket = flag.String("unix_socket", "", "vtgate unix socket")
-	protocol   = flag.String("protocol", "mysql", "client protocol, either mysql (default), grpc-vtgate, or grpc-vttablet")
-	user       = flag.String("user", "", "username to connect using mysql (password comes from the db-credentials-file)")
-	db         = flag.String("db", "", "db name to use when connecting / running the queries (e.g. @replica, keyspace, keyspace/shard etc)")
-
-	// test flags
-	deadline = flag.Duration("deadline", 5*time.Minute, "maximum duration for the test run (default 5 minutes)")
-	sql      = flag.String("sql", "", "sql statement to execute")
-	threads  = flag.Int("threads", 2, "number of parallel threads to run")
-	count    = flag.Int("count", 1000, "number of queries per thread")
+	host, unixSocket, user, db, sql string
+	port                            int
+	protocol                        = "mysql"
+	deadline                        = 5 * time.Minute
+	threads                         = 2
+	count                           = 1000
 )
 
-func main() {
-	logger := logutil.NewConsoleLogger()
-	flag.CommandLine.SetOutput(logutil.NewLoggerWriter(logger))
+func initFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&host, "host", "", "vtgate host(s) in the form 'host1,host2,...'")
+	fs.IntVar(&port, "port", 0, "vtgate port")
+	fs.StringVar(&unixSocket, "unix_socket", "", "vtgate unix socket")
+	fs.StringVar(&protocol, "protocol", protocol, "client protocol, either mysql (default), grpc-vtgate, or grpc-vttablet")
+	fs.StringVar(&user, "user", "", "username to connect using mysql (password comes from the db-credentials-file)")
+	fs.StringVar(&db, "db", "", "db name to use when connecting / running the queries (e.g. @replica, keyspace, keyspace/shard etc)")
 
-	defer exit.Recover()
+	fs.DurationVar(&deadline, "deadline", deadline, "maximum duration for the test run (default 5 minutes)")
+	fs.StringVar(&sql, "sql", "", "sql statement to execute")
+	fs.IntVar(&threads, "threads", threads, "number of parallel threads to run")
+	fs.IntVar(&count, "count", count, "number of queries per thread")
```

Comment:
We should use the variable itself for the default value. For example `fs.IntVar(&port, "port", port, "vtgate port")`

Reply:
I was thinking that since the default value is 0 for both the variable and the flag I might as well use a literal, just like we do for string flags

Reply:
see my comment [here](https://github.com/vitessio/vitess/pull/11364#discussion_r982233227) (we should _also_ not use literals for the string flags)